docs/Stream-API.md

```diff
@@ -0,0 +1,473 @@
+---
+title: Stream API Overview
+layout: documentation
+documentation: true
+---
+
+* [Concepts](#concepts)
+    * [Stream Builder] (#streambuilder)
```

Comment:
Some of the IDE MD plugins and tools are not generating links appropriately when there is a space. You can see the syntax [here](https://guides.github.com/features/mastering-markdown/) and [here] (https://en.support.wordpress.com/markdown-quick-reference/)

Even Github comments throws broken links when there is space between [] and (), you can observe above `here` links to see the difference. I guess it is better to address that.